fpsd/features.py

```diff
@@ -0,0 +1,765 @@
+                 "SELECT exampleid,                                 "
+                 "  avg( difference ) as mean_intercell_time,       "
+                 "  stddev( difference )                            "
+                 "  as standard_deviation_intercell_time            "
+                 "FROM interpacket_times GROUP BY exampleid);       ")
+        self.execute_query_from_string(query)
+
+        return "features.interpacket_timings"
+
+    def generate_table_initial_cell_directions(self, num_cells=10):
+        """This method takes all examples and produces a
+        table features.initial_cell_directions with the following format:
+
+        exampleid  | a         | b         ...    | num_cells
+        (integer)  | (integer) | (integer) ...    | (integer)
+        ------------------------------------------------------
+        9          | 0         | 1         ...
+        10         | 1         | 1         ...
+        11         | 1         | 0         ...
+
+        where there are a variable number (num_cells) of
+        feature columns, each named direction_cell_x where x is the
+        position in the trace, beginning at 1. This feature represents
+        the direction (where outgoing is positive) of the cell.
+
+        Args:
+            num_cells [int]: number of initial cells to use as features
+
+        Returns:
+            [string] name of newly created table
+        """
+
+        self.drop_stale_feature_table("features.initial_cell_directions")
+
+        crosstab_columns = ['direction_cell_{} integer'.format(x+1)
+                            for x in range(num_cells)]
+        query = ("CREATE TABLE features.initial_cell_directions AS  "
+                 "  (SELECT *                                       "
+                 "  FROM crosstab(                                  "
+                 "    'SELECT                                       "
+                 "    exampleid, rank,                              "
+                 "     case when ingoing = true then 0 else 1 end   "
+                 "  FROM (                                          "
+                 "    SELECT                                        "
+                 "      ROW_NUMBER() OVER                           "
+                 "      (PARTITION BY exampleid ORDER BY t_trace)   "
+                 "      AS rank,                                    "
+                 "      t.*                                         "
+                 "    FROM raw.frontpage_traces t) x                "
+                 "  WHERE x.rank <= 10')                            "
+                 "  AS  ct(exampleid integer, {})                   "
+                 ");").format(', '.join(crosstab_columns))
+
+        self.execute_query_from_string(query)
+        return "features.initial_cell_directions"
+
+    def generate_table_outgoing_cell_ordering(self, num_features=500):
```

Comment:
* Okay changed this to `num_cells` instead of `num_features` since it is more informative in describing the feature. 

* Yep this is true - if there are not 500 outgoing cells (which there aren’t in the vast majority of cases) then the missing features are Null. 

* Alright, replaced `ordering` with `positions` in these table names